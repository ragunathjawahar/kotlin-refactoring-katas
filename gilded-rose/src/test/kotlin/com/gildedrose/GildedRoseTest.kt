package com.gildedrose

import java.io.ByteArrayOutputStream
import java.io.PrintStream
import org.approvaltests.Approvals
import org.junit.jupiter.api.Test

class GildedRoseTest {
  @Test
  fun `basic test for 2 days`() {
    // given
    val outStream = ByteArrayOutputStream()
    System.setOut(PrintStream(outStream))

    // when
    main(emptyArray())

    // then
    Approvals.verify(outStream.toString())
  }
}
