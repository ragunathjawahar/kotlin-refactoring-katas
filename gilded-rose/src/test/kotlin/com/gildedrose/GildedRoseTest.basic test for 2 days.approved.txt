OMGHAI!
-------- day 0 --------
name, sellIn, quality
+5 Dexterity Vest, 10, 20
Aged Brie, 2, 0
Elixir of the Mongoose, 5, 7
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 15, 20
Backstage passes to a TAFKAL80ETC concert, 10, 49
Backstage passes to a TAFKAL80ETC concert, 5, 49
Conjured Mana Cake, 3, 6

-------- day 1 --------
name, sellIn, quality
+5 Dexterity Vest, 9, 19
Aged Brie, 1, 1
Elixir of the Mongoose, 4, 6
Sulfuras, Hand of Ragnaros, 0, 80
Sulfuras, Hand of Ragnaros, -1, 80
Backstage passes to a TAFKAL80ETC concert, 14, 21
Backstage passes to a TAFKAL80ETC concert, 9, 50
Backstage passes to a TAFKAL80ETC concert, 4, 50
Conjured Mana Cake, 2, 5

